### PR TITLE
Use format_string() in core/git_access.c

### PR DESCRIPTION
**Pull request by bstoeger**
Since this helper-function exists, we might just use it.

Signed-off-by: Berthold Stoeger <bstoeger@mail.tuwien.ac.at>

### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [x] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

This is a very small code-simplification: use the `format_string()` helper instead of determining size + `malloc()`ing by hand.
Two comments:
1) I did not test all code paths, since this would mean making a fresh cloud repository.
2) I'm not very happy about the "unsafe" `sprintf()`. Sure, we know that this works, but it somehow leaves a bad aftertaste. Perhaps replace this by a `format_string()` as well, even if this means an additional memory allocation?

### CODE REVIEW

File: core/git-access.c [hunk 1]
```diff
@@ -692,15 +691,11 @@ static git_repository *create_and_push_remote(const char *localdir, const char *
...
-	variable_name = malloc(len);
-	snprintf(variable_name, len, "branch.%s.remote", branch);
+	variable_name = format_string("branch.%s.remote", branch);
 	git_config_set_string(conf, variable_name, "origin");
 	/* we know this is shorter than the previous one, so we reuse the variable*/
-	snprintf(variable_name, len, "branch.%s.merge", branch);
-	len = sizeof("refs/heads/") + strlen(branch);
-	merge_head = malloc(len);
-	snprintf(merge_head, len, "refs/heads/%s", branch);
+	sprintf(variable_name, "branch.%s.merge", branch);
```
Comment (neolit123) [inline core/git-access.c]: i would slightly change the comment to be more verbose:
`/* "branch.%s.remote" is longer than "branch.%s.merge", so we can re-use the buffer. */`.

like you suggest, the clearer path for the reader would be to allocate a second buffer and free both at the end, but i don't think its that necessary. what we have is less readable, but functional as long as `strlen("remote") > strlen("merge")`.

**Review by neolit123 — APPROVED** [review]

**Review by neolit123 — CHANGES_REQUESTED** [review]
requesting a small change for the comment about `sprintf`. other than that lgtm.

File: core/git-access.c (hunk 1 above)
Comment (bstoeger) [reply core/git-access.c]: Good point - changed that.

**Review by neolit123 — APPROVED** [review]

File: core/git-access.c [hunk 2]
```diff
@@ -692,15 +691,11 @@ static git_repository *create_and_push_remote(const char *localdir, const char *
...
-	snprintf(variable_name, len, "branch.%s.remote", branch);
+	variable_name = format_string("branch.%s.remote", branch);
 	git_config_set_string(conf, variable_name, "origin");
-	/* we know this is shorter than the previous one, so we reuse the variable*/
-	snprintf(variable_name, len, "branch.%s.merge", branch);
-	len = sizeof("refs/heads/") + strlen(branch);
-	merge_head = malloc(len);
-	snprintf(merge_head, len, "refs/heads/%s", branch);
+	/* "branch.%s.remote" is longer than "branch.%s.merge", so we can re-use the buffer. */
+	sprintf(variable_name, "branch.%s.merge", branch);
```
Comment (dirkhh) [inline core/git-access.c]: optimizations that create code that just Not Good Code(tm) worry me.
I'd rather use another format_string here...

**Review by dirkhh — CHANGES_REQUESTED** [review]

File: core/git-access.c (hunk 2 above)
Comment (bstoeger) [reply core/git-access.c]: OK - adapted commit and commit-message.

**Comment by neolit123** [comment]
thanks @bstoeger

**Pull request merged**